src/vec_int16_ppc.h

```diff
@@ -61,9 +79,447 @@
+ * \endcode
+ * The vmulouh, vmuleuh, vperm instruction sequence is the core of the
+ * function.  They multiply the elements and selects/merges the high
+ * order 16-bits of each product into the result vector.
+ *
+ * \section int16_examples_0_1 Examples, Divide by integer constant
+ * Suppose we have a requirement to convert an array of 16-bit
+ * unsigned short values to decimal. The classic <I>itoa</I>
+ * implementation performs a sequence of divide / modulo by 10
+ * operations that produce on (decimal) per iteration,
+ * until the divide returns 0.
+ *
+ * For this example we want to vectorize the operation and the
+ * PowerISA (and and most other platforms) does not provide a
+ * vector integer divide instruction. But we do have vector integer
+ * multiply as we will see the multiply high defined above is very
+ * handy for applying the multiplicative inverse.
+ * Also we the conversion divide is a constant value applied across
+ * the vector which simplifies the coding.
+ *
+ * Here we can use the multiplicative inverse which is a
+ * scaled fixed point fraction calculated from the original divisor.
+ * This works nicely if the fixed radix point is just before
+ * the 16-bit fraction and we have a multiply high (vec_mulhuh())
+ * operation. Multiplying a 16-bit unsigned integer by a 16-bit
+ * unsigned fraction generates a 32-bit product with 16-bits above
+ * (integer) and below (fraction) the radix point. The high 16-bits
+ * of the product is a good approximation of the integer quotient.
+ *
+ * It turns out that generating the multiplicative inverse can be
+ * tricky.  To produce correct results over the full range, requires
+ * possible pre-scaling and post-shifting, and sometimes a corrective
+ * addition. Fortunately the mathematics are well
+ * understood and are commonly used in optimizing compilers.
+ * Even better, Henry Warren's book has a whole chapter on this topic.
+ * \see "Hacker's Delight, 2nd Edition,"
+ * Henry S. Warren, Jr, Addison Wesley, 2013.
+ * Chapter 10, Integer Division by Constants.
+ *
+ * \subsection int16_examples_0_1_1 Divide by constant 10 examples
+ * In the chapter above;
+ * <BLOCKQUOTE>Figure 10-2 Computing the magic number for unsigned division.</BLOCKQUOTE>
+ * provides a sample C function for generating the magic number
+ * (actually a struct containing; the magic multiplicative inverse,
+ * "add" indicator, and the shift amount.).
+ * For the 16-bit unsigned divisor 10 this is {  52429, 0, 3 }:
+ * - the multiplier is 52429.
+ * - no corrective add of the dividend is required.
+ * - the final shift is 3-bits right.
+ *
+ * Which could look like this:
+ * \code
+ vui16_t
+__test_div10 (vui16_t n)
+{
+  vui16_t q;
+  // M= 52429, a=0, s=3
+  vui16_t magic = vec_splats ((unsigned short) 52429);
+  const int s = 3;
+
+  q = vec_mulhuh (magic, n);
+  return vec_srhi (q, s);
+}
+ * \endcode
+ *
+ * But we need also need the modulo to extract each digit. This simplest and oldest technique is to
```

Comment:
s/need also need/also need/

Reply:
Ok, also changed the s/This simplest/The simplest/